### PR TITLE
lockExpire set by configuration

### PR DESCRIPTION
**Pull request by KihwanCheon**
Hi.
This is following  [#306](https://github.com/magro/memcached-session-manager/issues/306)

I modify lock expire in seconds.
Source code is based on version 1.8.3 that my service use.

### CODE REVIEW

File: core/src/main/java/de/javakaffee/web/msm/MemcachedSessionService.java [hunk 1]
```diff
@@ -1633,10 +1635,24 @@ public long getOperationTimeout() {
 		return _operationTimeout;
 	}
 
+    /**
+     * The expire in seconds after that a session backup is locked automatically works by memcached
```
Comment (magro) [inline core/src/main/java/de/javakaffee/web/msm/MemcachedSessionService.java]: Should it be "unlocked" instead of "locked"?

File: core/src/main/java/de/javakaffee/web/msm/MemcachedSessionService.java [hunk 2]
```diff
@@ -447,6 +448,7 @@ void startInternal() throws LifecycleException {
                 "\n- node ids: " + _memcachedNodesManager.getPrimaryNodeIds() +
                 "\n- failover node ids: " + _memcachedNodesManager.getFailoverNodeIds() +
                 "\n- storage key prefix: " + _memcachedNodesManager.getStorageKeyFormat().prefix +
+                "\n- locking mode: " + _lockingMode + "(expire " + _lockExpire + "s)" +
```
Comment (magro) [inline core/src/main/java/de/javakaffee/web/msm/MemcachedSessionService.java]: Add space before `(expire`?

File: core/src/main/java/de/javakaffee/web/msm/MemcachedSessionService.java [hunk 3]
```diff
@@ -1633,10 +1635,24 @@ public long getOperationTimeout() {
 		return _operationTimeout;
 	}
 
+    /**
+     * The expire in seconds after that a session backup is locked automatically works by memcached
+     * when {@link #getLockingStrategy()} is <code>{@link LockingMode#AUTO} or {@link LockingMode#ALL}</code>.
+     * If {@link #getOperationTimeout()} is less then this, Other request is available session backup
+     * when before request that processed by other tomcat delayed over {@link #getOperationTimeout()}.
```
Comment (magro) [inline core/src/main/java/de/javakaffee/web/msm/MemcachedSessionService.java]: Can you rephrase this? (I don't understand it)

**Comment by magro** [comment]
Thanks for the pull request! Some questions:
* Can you rename `lockExpire` to `lockExpiration`?
* Can you describe the rules that `lockExpiration` should follow in combination with `operationTimeout`? (e.g. which one should be bigger than the other one?)
* Perhaps log a warning if on startup it's detected that this rule is broken?
* Can you describe what's happening (or might happen) when this rule is broken?

Thanks a lot in advance!

File: core/src/main/java/de/javakaffee/web/msm/MemcachedSessionService.java (hunk 1 above)
Comment (KihwanCheon) [inline core/src/main/java/de/javakaffee/web/msm/MemcachedSessionService.java]: Yes, It is better.

**Comment by KihwanCheon** [comment]
This needed when request processing time is over 5 seconds(before lockExpiration).
I think lockExpiration greater then operationTimeout or not is able to choice.
I use non-sticky tomcat cluster. Clients App retry same request over some seconds.
So I extend OperationTimeout, but not works request processing time is over than 5 seconds. 
(It's problem that processing time is over 5 seconds. :(  However I need avoid duplicated processing.)

I do test and write result below.

* lockExpiration > OperationTimeout : 
  - The other request wait for lockExpiration or unlocking.
  - The other request wait for as much as OperationTimeout, after then backup session used as LockStatus.COULD_NOT_AQUIRE_LOCK.

* lockExpiration < OperationTimeout : 
  - The other request wait for as much as lockExpiration, that is accepted in lockExpiration. 
  - The other request don't wait, that is accepted after lockExpiration.

**Comment by magro** [comment]
@KihwanCheon Sorry for the late response, and thanks for the explanation! I just resolved conflicts with the current master and just merged it. Thanks!
